test/soak/serve_hostnames/serve_hostnames.go

```diff
@@ -120,27 +141,36 @@ func main() {
 		for j := 0; j < *perNode; j++ {
 			podName := fmt.Sprintf("serve-hostname-%d-%d", i, j)
 			podNames = append(podNames, podName)
-			glog.Infof("Creating pod %s/%s on node %s", ns, podName, node.Name)
-			_, err := c.Pods(ns).Create(&api.Pod{
-				ObjectMeta: api.ObjectMeta{
-					Name: podName,
-					Labels: map[string]string{
-						"name": "serve-hostname",
+			// Make several attempts
```

Comment:
As soon as I saw that you'd added the retries inside this loop it made me think of running it in parallel. If you start to have failures creating pods, then the setup step could start to take a really long time. I don't think it would be terribly difficult to make the output clear (you could have a progress counter that went from 0 to N as each pod was created and then output all of the latencies, sorted if you like, at the end). But totally optional.